APCAppCore/APCAppCore/UI/TabBarControllers/Activities/APCActivitiesViewController.m

```diff
@@ -131,53 +139,46 @@ - (void) viewWillAppear: (BOOL) animated
         self.showingConsentFlow = NO;
         [self showReconsentIfNecessary];
     }
-    else {
-        [self reloadData];
-    }
     
     [self setUpNavigationBarAppearance];
     
     APCLogViewControllerAppeared();
 }
 
 - (void) reloadData {
-    [self setupNotifications];
     [self reloadTasksFromCoreData];
     [self checkForAndMaybeRespondToSystemDateChange];
 }
 
-- (void) viewDidDisappear: (BOOL) animated
-{
-    [super viewDidDisappear: animated];
-
-    [self cancelNotifications];
-}
-
 - (void) setupNotifications
 {
     // Fires when one day rolls over to the next.
     [[NSNotificationCenter defaultCenter] addObserver: self
-                                             selector: @selector (checkForAndMaybeRespondToSystemDateChange)
+                                             selector: @selector (handleSystemDateChangeNotification)
                                                  name: APCDayChangedNotification
                                                object: nil];
 
     // ...but that only happens every minute or so.  This lets us respond much faster.
     [[NSNotificationCenter defaultCenter] addObserver: self
-                                             selector: @selector (checkForAndMaybeRespondToSystemDateChange)
+                                             selector: @selector (handleSystemDateChangeNotification)
                                                  name: UIApplicationWillEnterForegroundNotification
                                                object: nil];
 
     // ...but that only happens every minute or so.  This lets us respond much faster.
     [[NSNotificationCenter defaultCenter] addObserver: self
-                                             selector: @selector (checkForAndMaybeRespondToSystemDateChange)
+                                             selector: @selector (handleSystemDateChangeNotification)
                                                  name: UIApplicationDidBecomeActiveNotification
                                                object: nil];
+    
+    if (self.appDelegate.promptUserToContinueActivities) {
```

Comment:
Indeed. Had this test here before changing how data is loaded. Thx for the catch. PR updated